connection.js

```diff
@@ -650,532 +651,234 @@ class Connection {
+        // then we process the UHMAEATs for periodtimes blitz/live/corr
+        // and reject challenge if not within min/max set limits by bot admin :
+        let resultPeriodtimeBlitz = UHMAEATForPeriodtimes ("blitz");
+        if (resultPeriodtimeBlitz) {
+            return (resultPeriodtimeBlitz);
+        }
+
+        let resultPeriodtimeLive = UHMAEATForPeriodtimes ("live");
+        if (resultPeriodtimeLive) {
+            return (resultPeriodtimeLive);
+        }
+
+        let resultPeriodtimeCorr = UHMAEATForPeriodtimes ("corr");
+        if (resultPeriodtimeCorr) {
+            return (resultPeriodtimeCorr);
+        }
+        // end of periodtime rejects
+
+        return { reject: false };  // Ok !
+
+        /////////////////////////////////////////////////////////////////////////////////////
+        // before reading UHMAEATs function code, general information : 
+        // 0) "none" doesnt have a period time, so we let it slide from both maintime and 
+        // periodtime rejects
+        // (we have sanity checks just in case)
+        // 1) simple time doesn't have a main time, only a period time, so we let it slide 
+        // from maintime rejects
+        // 2) fischer : doesnt have a minperiods or maxperiods (blitz/live/corr)
+        // 3) absolute doesnt have a period time, so we let it slide from periodtime rejects
+        // 4) while using gedit "find and replace", make sure you dont replace
+        // t.max_time to t.min_time ! (it doesnt exist !)
+        //////////////////////////////////////////////////////////////////////////////////////
+
+        function UHMAEATForPeriodtimes (blitzLiveCorr) {
+            ////// begining of *** UHMAEAT v3.0: Universal Highly Modulable And Expandable Argv Tree ***
+            ///// version 3.0 for periodtimes
+        
+            let minGeneralArg = "min" + "periodtime" + blitzLiveCorr; 
+            // for example "minperiodtimeblitz"
+            let minFamilyArray = [];
+            minFamilyArray.push(minGeneralArg);
+            minFamilyArray.push(minGeneralArg + "ranked");
+            minFamilyArray.push(minGeneralArg + "unranked");
+            // example : ["minperiodtimeblitz", "minperiodtimeblitzranked", "minperiodtimeblitzunranked"];
+
+            let maxGeneralArg = "max" + "periodtime" + blitzLiveCorr; 
+            // for example "maxperiodtimeblitz"
+            let maxFamilyArray = [];
+            maxFamilyArray.push(maxGeneralArg);
+            maxFamilyArray.push(maxGeneralArg + "ranked");
+            maxFamilyArray.push(maxGeneralArg + "unranked");
+            // example : ["maxperiodtimeblitz", "maxperiodtimeblitzranked", "maxperiodtimeblitzunranked"];
```

Comment:
Instead of 
```
let maxFamilyArray = [];
maxFamilyArray.push(maxGeneralArg);
maxFamilyArray.push(maxGeneralArg + "ranked");
maxFamilyArray.push(maxGeneralArg + "unranked");
```
you can do 
```
let maxFamilyArray = [maxGeneralArg, maxGeneralArg + "ranked", maxGeneralArg + "unranked"];
```
or
```
maxFamilyArray = ["", "ranked", "unranked"].map(e => maxGeneralArg + e);
```